src/main/lrsql/util/statement.clj

```diff
@@ -9,16 +9,54 @@
 
 ;; If a Statement lacks a version, the version MUST be set to 1.0.0
 ;; TODO: Change for version 2.0.0
+;; FIXME: Why is this not version 1.0.3?
 (def xapi-version "1.0.0")
 
 ;; NOTE: SQL LRS overwrites any pre-existing authority object in a statement, as
 ;; suggested by the spec:
 ;; https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#requirements-14
 
+(defn- dissoc-empty-lang-maps*
+  [{:strs [display name description] :as m}]
+  (cond-> m
+    (empty? display)     (dissoc "display")
+    (empty? name)        (dissoc "name")
+    (empty? description) (dissoc "description")))
+
+(defn- dissoc-empty-lang-maps
+  [statement]
+  (cond-> statement
+    ;; Dissoc empty verb display
+    true
+    (update "verb" dissoc-empty-lang-maps*)
+    ;; Dissoc empty object activity name + description
+    (= "Activity" (get-in statement ["object" "objectType"]))
+    (update-in ["object" "definition"]
```

Comment:
@kelvinqian00 isn't this too specific? in my dealings with this in the past empty language maps will not pass in if they appear anywhere in a statement. Limiting it to `object.definition` and `attachments` will only allow  cover those two cases, and if the incoming statements have lmaps at other points in the map (for example: `verb.display`), those will not pass through.

@cliffcaseyyet correct me if I'm wrong but we have no way to know if there will or won't be empty lmaps in the statement, and if were too specific we run the risk of making edge cases that fall outside the spec in weird ways.

Reply:
The problem with that is that there are places where empty maps may actually make a semantic difference, specifically in extension values. My first approach was an implementation that generically removed empty maps via `w/postwalk`, caused DATASIM-based tests to fail (since DATASIM likes to put all sorts of values, empty maps included, in extension values), so I changed it down to hyper-specific xAPI property removal.

Reply:
Correct me if i'm wrong @kelvinqian00 but isn't that what  `dissoc-empty-lang-maps*` is about? Looks like definition was an edge case where it didn't have a common lmap kw.

Reply:
If you look at the xAPI spec, there are three names for language map properties: `display` (verbs and attachments), `name` (activities), and `description` (activities and attachments). That is what `dissoc-empty-lang-maps*` covers.